go/vt/vitessdriver/driver.go

```diff
@@ -225,25 +182,29 @@ func (c *conn) dial() error {
 	return err
 }
 
-func (c *conn) useExecuteShards() bool {
-	return c.Keyspace != "" && c.Shard != ""
-}
-
 func (c *conn) Prepare(query string) (driver.Stmt, error) {
 	return &stmt{c: c, query: query}, nil
 }
 
+func (c *conn) PrepareContext(ctx context.Context, query string) (driver.Stmt, error) {
+	return c.Prepare(query)
+}
+
 func (c *conn) Close() error {
 	c.vtgateConn.Close()
 	return nil
 }
 
 func (c *conn) Begin() (driver.Tx, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), c.Timeout)
+	defer cancel()
+	return c.BeginContext(ctx)
+}
+
+func (c *conn) BeginContext(ctx context.Context) (driver.Tx, error) {
 	if c.Streaming {
 		return nil, errors.New("transaction not allowed for streaming connection")
 	}
-	ctx, cancel := context.WithTimeout(context.Background(), c.Timeout)
-	defer cancel()
 	tx, err := c.vtgateConn.Begin(ctx)
```

Comment:
If there is an explicit iso set, you can define your own values in the driver for the client to use, along with a WithContext function. If it is implicit, then up to you...

The driver package requests drivers to check for the presence of both of them and return an error if not supported. But if you don't want to do that, just add a line in the doc.